sympy/tensor/array/nditer.py

```diff
@@ -0,0 +1,58 @@
+from sympy.tensor.array.dense_ndim_array import DenseNDimArray
+from sympy.tensor.array.sparse_ndim_array import SparseNDimArray
+from sympy.tensor.array.ndim_array import NDimArray
+from sympy.core.compatibility import Iterable
+
+class nditer(object):
+
+    def __init__(self, iterable):
+        from sympy.matrices.matrices import MatrixBase
+
+        if not isinstance(iterable, (Iterable, MatrixBase)):
+            raise NotImplementedError("Data type not yet supported")
+        if isinstance(iterable, list) and isinstance(iterable[0], NDimArray):
+            temp = []
+            for i in range(len(iterable)):
+                temp += iterable[0].tolist()
+            iterable = temp
+        self._iter = iterable
+        self._idx = 0
+
+    def __iter__(self):
+        return self
```

Comment:
why not return the iterator here?

Reply:
I was thinking the `NDIter` itself as an iterator, that is why `self` is returned here.